contrib/pinot-druid-migration/src/main/java/org/apache/pinot/druid/tools/hadoop/DruidToPinotSegmentConverterHadoopJob.java

```diff
@@ -0,0 +1,438 @@
+import org.apache.pinot.common.utils.StringUtil;
+import org.apache.pinot.common.utils.TarGzCompressionUtils;
+import org.apache.pinot.controller.helix.ControllerRequestURLBuilder;
+import org.apache.pinot.core.indexsegment.generator.SegmentVersion;
+import org.apache.pinot.hadoop.job.BaseSegmentJob;
+import org.apache.pinot.hadoop.job.ControllerRestApi;
+import org.apache.pinot.hadoop.job.DefaultControllerRestApi;
+import org.apache.pinot.hadoop.job.JobConfigConstants;
+import org.apache.pinot.hadoop.utils.JobPreparationHelper;
+import org.apache.pinot.hadoop.utils.PushLocation;
+
+
+public class DruidToPinotSegmentConverterHadoopJob extends BaseSegmentJob {
+  protected static final String APPEND = "APPEND";
+
+  protected final Path _inputPattern;
+  protected final Path _outputDir;
+  protected final Path _stagingDir;
+  protected final String _rawTableName;
+  protected final List<PushLocation> _pushLocations;
+
+  // Optional
+  protected final Path _depsJarDir;
+  protected final Path _schemaFile;
+  protected final String _defaultPermissionsMask;
+
+  // Output Directory FileSystem
+  protected FileSystem _outputDirFileSystem;
+
+  public DruidToPinotSegmentConverterHadoopJob(Properties properties) {
+    super(properties);
+    _conf.set("mapreduce.job.user.classpath.first", "true");
+    _inputPattern = Preconditions.checkNotNull(getPathFromProperty(JobConfigConstants.PATH_TO_INPUT));
+    _outputDir = Preconditions.checkNotNull(getPathFromProperty(JobConfigConstants.PATH_TO_OUTPUT));
+    _stagingDir = new Path(_outputDir, UUID.randomUUID().toString());
+    _rawTableName = Preconditions.checkNotNull(_properties.getProperty(JobConfigConstants.SEGMENT_TABLE_NAME));
+
+    // Optional
+    _depsJarDir = getPathFromProperty(JobConfigConstants.PATH_TO_DEPS_JAR);
+    _schemaFile = getPathFromProperty(JobConfigConstants.PATH_TO_SCHEMA);
+    _defaultPermissionsMask = _properties.getProperty(JobConfigConstants.DEFAULT_PERMISSIONS_MASK);
+
+    // Optional push location and table parameters. If set, will use the table config and schema from the push hosts.
+    String pushHostsString = _properties.getProperty(JobConfigConstants.PUSH_TO_HOSTS);
+    String pushPortString = _properties.getProperty(JobConfigConstants.PUSH_TO_PORT);
+    if (pushHostsString != null && pushPortString != null) {
+      _pushLocations =
+          PushLocation.getPushLocations(StringUtils.split(pushHostsString, ','), Integer.parseInt(pushPortString));
+    } else {
+      _pushLocations = null;
+    }
+
+    _logger.info("*********************************************************************");
+    _logger.info("Input Pattern: {}", _inputPattern);
+    _logger.info("Output Directory: {}", _outputDir);
+    _logger.info("Staging Directory: {}", _stagingDir);
+    _logger.info("Raw Table Name: {}", _rawTableName);
+    _logger.info("*********************************************************************");
+  }
+
+  protected boolean isDataFile(URI s) {
```

Comment:
Maybe we can add a quick unit test for this method since it's a little tricky

Reply:
Do I just add another test class to the test folder?

Reply:
yes :) just for this method if possible